File: Aufgabe3/Javascript/a3_6.js

function f1() {
  var a = 42;
  var x = "Bye";

  function f2(x) {
    a = "Hello";
    x = 27;

    function f3(a, x) {
      console.log(x);
      x = 3.14;
    }

    f3(x, a);
  }

  f2(x);
  console.log(a);
  console.log(x);
}

f1();

File: Aufgabe3/Python/a3_6.py

def f1():
  a = 42
  x = "Bye"

  def f2(x):
    a = "Hello"
    x = 27

    def f3(a, x):
      print(x)
      x = 3.14

    f3(x, a)

  f2(x)
  print(a)
  print(x)

f1()
